Packages/io.chainsafe.web3-unity/Runtime/Scripts/EVM/IPFS/IPFS.cs

```diff
@@ -163,6 +174,12 @@ private static async Task<string> Upload(string apiKey, string bucketId, string
             {
                 throw new WebException(requestUpload.error);
             }
+            
+            var uploadResponse = JsonConvert.DeserializeObject<Path>(requestUpload.downloadHandler.text);
+            if (uploadResponse.files_details[0].status == "failed")
+            {
+                throw new Exception($"Upload Failed: {uploadResponse.files_details[0].message}");
```

Comment:
We can throw Web3Exception here I think as it's specific to our SDK.

Reply:
good idea